lib/SILOptimizer/Mandatory/SendNonSendable.cpp

```diff
@@ -0,0 +1,33 @@
+#include "../../Sema/TypeCheckConcurrency.h"
+#include "swift/AST/Expr.h"
+#include "swift/SILOptimizer/PassManager/Transforms.h"
+
+using namespace swift;
+
+class SendNonSendable : public SILFunctionTransform {
+
+  // find any ApplyExprs in this function, and check if any of them make an
+  // unsatisfied isolation jump, emitting appropriate diagnostics if so
+  void run() override {
+    SILFunction *function = getFunction();
+
+    for (SILBasicBlock &bb : *function) {
+      for (SILInstruction &instr : bb) {
+        if (ApplyExpr *apply = instr.getLoc().getAsASTNode<ApplyExpr>()) {
+          if (apply->makesUnsatisfiedIsolationJump()) {
+            auto &ctx = instr.getModule().getASTContext();
+            if (ctx.LangOpts.hasFeature(Feature::DeferredSendableChecking)) {
```

Comment:
Can this feature check move way out? Either don't add the pass at all, or early-exit from the pass, if it's not set?